File: salesforge/backend/app/schemas/lead.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class LeadCreate(BaseModel):
    first_name: str
    last_name: str
    email: str | None = None
    phone: str | None = None
    company_name: str | None = None
    status: str = "new"
    source: str | None = None
    notes: str | None = None


class LeadUpdate(BaseModel):
    first_name: str | None = None
    last_name: str | None = None
    email: str | None = None
    phone: str | None = None
    company_name: str | None = None
    status: str | None = None
    source: str | None = None
    notes: str | None = None


class LeadResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    first_name: str
    last_name: str
    email: str | None
    phone: str | None
    company_name: str | None
    status: str
    source: str | None
    notes: str | None
    owner_id: str
    created_at: datetime
    updated_at: datetime


class LeadListResponse(BaseModel):
    items: list[LeadResponse]
    total: int
    page: int
    page_size: int


class LeadConvertRequest(BaseModel):
    create_company: bool = True
    create_deal: bool = False
    deal_title: str | None = None
    deal_value: float | None = None


class LeadConvertResponse(BaseModel):
    contact_id: str
    company_id: str | None
    deal_id: str | None

File: salesforge/frontend/src/types/lead.ts
export type LeadStatus = 'new' | 'contacted' | 'qualified' | 'converted' | 'lost'

export interface Lead {
  id: string
  first_name: string
  last_name: string
  email: string | null
  phone: string | null
  company_name: string | null
  status: LeadStatus
  source: string | null
  notes: string | null
  owner_id: string
  created_at: string
  updated_at: string
}

export interface LeadListResponse {
  items: Lead[]
  total: number
  page: number
  limit: number
}

export interface LeadCreate {
  first_name: string
  last_name: string
  email?: string
  phone?: string
  company_name?: string
  source?: string
  notes?: string
}

export interface LeadUpdate {
  first_name?: string
  last_name?: string
  email?: string
  phone?: string
  company_name?: string
  source?: string
  notes?: string
  status?: LeadStatus
}

export interface LeadConvertRequest {
  create_company: boolean
  create_deal: boolean
  deal_title?: string
  deal_value?: number
}

export interface LeadConvertResponse {
  contact_id: string
  company_id: string | null
  deal_id: string | null
}
